src/v/cluster/tx_registry_frontend.cc

```diff
@@ -61,14 +61,49 @@ tx_registry_frontend::tx_registry_frontend(
       config::shard_local_cfg().metadata_dissemination_retry_delay_ms.value()) {
 }
 
+ss::future<bool> tx_registry_frontend::ensure_tx_topic_exists() {
+    if (_metadata_cache.local().contains(model::tx_manager_nt)) {
+        co_return true;
+    }
+
+    if (!co_await try_create_tx_topic()) {
+        vlog(clusterlog.warn, "failed to create {}", model::tx_manager_nt);
+        co_return false;
+    }
+
+    auto retries = _metadata_dissemination_retries;
+    auto delay_ms = _metadata_dissemination_retry_delay_ms;
+    std::optional<std::string> error;
+    while (!_as.abort_requested() && 0 < retries--) {
+        auto is_cache_filled = _metadata_cache.local().contains(
+          model::tx_manager_nt);
+        if (unlikely(!is_cache_filled)) {
+            error = vformat(
+              fmt::runtime("can't find {} in the metadata_cache cache"),
+              model::tx_manager_nt);
+            vlog(
+              clusterlog.trace,
+              "waiting for {} to fill metadata_cache cache, retries left: {}",
+              model::tx_manager_nt,
+              retries);
+            co_await sleep_abortable(delay_ms, _as);
+            continue;
+        }
+        co_return true;
+    }
```

Comment:
It requires wiring `controller_api` into `tx_registry_frontend`; also wait_for_topic is more complex so I couldn't verify on the spot that it does the same (wait until metadata_cache is filled). The loop does the job so it's ok.

Reply:
i am asking about this as here we duplicate code doing almost the same thing

Reply:
`wait_for_topics` is more robust as it guarantees that the topic is ready to be used

Reply:
curious whats the conclusion here.

Reply:
Synced offline, Michal agreed to merge it as is. `wait_for_topics` is more complex and requires additional wiring to do it as part of the fix. We'll do it later because current place isn't the only one which is worth switching to wait_for_topics.